AntennaTracker/control_auto.cpp

```diff
@@ -17,12 +17,44 @@ void Tracker::update_auto(void)
         return;
     }
 
-    float yaw = wrap_180_cd((nav_status.bearing+g.yaw_trim)*100) * 0.01f;
-    float pitch = constrain_float(nav_status.pitch+g.pitch_trim, -90, 90);
+    float yaw = wrap_180_cd((nav_status.bearing+g.yaw_trim)*100); // target yaw in centidegrees
+    float pitch = constrain_float(nav_status.pitch+g.pitch_trim, -90, 90) * 100; // target pitch in centidegrees
+
+    calc_angle_error(pitch, yaw);
+
+    float bf_pitch;
+    float bf_yaw;
+    calc_body_frame_target(pitch, yaw, bf_pitch, bf_yaw);
 
     // only move servos if target is at least distance_min away
     if ((g.distance_min <= 0) || (nav_status.distance >= g.distance_min)) {
-        update_pitch_servo(pitch);
-        update_yaw_servo(yaw);
+        update_pitch_servo(bf_pitch);
+        update_yaw_servo(bf_yaw);
     }
 }
+void Tracker::calc_angle_error( float pitch, float yaw)
+{
+	// Pitch angle error in centidegrees
+	// Positive error means the target is above current pitch
+    // Negative error means the target is below current pitch
+    float ahrs_pitch = ahrs.pitch_sensor;
+    nav_status.angle_error_pitch = pitch - ahrs_pitch;
+
+    // Yaw angle error in centidegrees
+	// Positive error means the target is right of current yaw
+    // Negative error means the target is left of current yaw
+    int32_t ahrs_yaw_cd = wrap_180_cd(ahrs.yaw_sensor);
+    nav_status.angle_error_yaw = wrap_180_cd(yaw - ahrs_yaw_cd);
+
+    // earth frame to body frame angle error conversion
+    float bf_pitch_err   = ahrs.cos_roll()     * nav_status.angle_error_pitch    + ahrs.sin_roll()  * ahrs.cos_pitch() * nav_status.angle_error_yaw;
```

Comment:
I think because these calculations are the same as those in the calc_body_frame_target function we could re-use that function here?